drivers/gpio/gpio_nrfx.c

```diff
@@ -590,26 +582,20 @@ static int gpio_nrfx_init(const struct device *port)
 	static const struct gpio_nrfx_cfg gpio_nrfx_p##id##_cfg = {	\
 		.common = {						\
 			.port_pin_mask =				\
-			GPIO_PORT_PIN_MASK_FROM_DT_NODE(GPIO(id)),	\
+			GPIO_PORT_PIN_MASK_FROM_DT_INST(id),		\
 		},							\
-		.port = NRF_P##id,					\
-		.port_num = id						\
+		.port = _CONCAT(NRF_P,DT_INST_PROP(id, port)),		\
```

Comment:
Perhaps the following will be a bit more readable:
```suggestion
		.port = (NRF_GPIO_Type *)DT_INST_REG_ADDR(id),		\
```

Reply:
done